src/cc.c

```diff
@@ -39,23 +39,26 @@ void* cc_muxp(int s, const void *a, const void *b)
 {
 	uintptr_t mask = -(s != 0);
 	uintptr_t ret = mask & (((uintptr_t)a) ^ ((uintptr_t)b));
-	ret = ret ^ ((uintptr_t)a);
-	return (void*) ret;
+    ret = ret ^ ((uintptr_t)b);
+    return (void*) ret;
 }
 
 /* https://cryptocoding.net/index.php/Coding_rules#Compare_secret_strings_in_constant_time */
 int cc_cmp_safe(size_t size, const void* a, const void* b)
 {
+    if ( size == 0) {
+        return 1;
+    }
```

Comment:
Looking at the implementation below, we're or'ing "characters at this position aren't equal", which to me sounds like it should return 0 for equal strings and 1 for otherwise, like `strcmp`. Then we shouldn't be returning 1 for two empty strings, and it should be perfectly handled by the generic codepath below. Where am I mistaken?

Reply:
I might misread it (and it wouldn't be the first time, English is not my 1st language) but the function documentation in the header file seems to say that it should return 1 if the strings are different or if the size passed as first argument is 0: https://github.com/darlinghq/darling-corecrypto/blob/master/include/corecrypto/cc.h#L83.

Reply:
Ah, I see, so the empty case is indeed explicitly defined to return 1. Then let's add a comment about it into the implementation too, because anyone reading this would have the same thought that I had :slightly_smiling_face: